### DISCUSSION TITLE
导入到滴答清单

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#4281] [插件说明]
描述
---

> 将正文导入到滴答清单（或 Ticktick）

地址
---

[https://simpread.ksria.cn/plugins/details/R1yEJZPj8s](https://simpread.ksria.cn/plugins/details/R1yEJZPj8s)

快捷键
---

`t c`

视频
---

https://user-images.githubusercontent.com/81074/181685752-7127e39e-0fe8-48fe-9938-81429ca725ec.mp4

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

![image](https://user-images.githubusercontent.com/81074/181676273-7be1c0d2-c20e-44f1-a874-55cc1a665636.png)

功能
---

1. 可使用简悦官方提供的 **Client ID 与 Secret** 也可用户自行设置。

2. 可选择 **滴答清单** 或 **Ticktick.com**

3. 导入正文到滴答清单

4. 添加标签，并可使用稍后读定制的标签

    ![image](https://user-images.githubusercontent.com/81074/181676504-4ab99a84-6724-43fc-9250-9f4a12a7d77a.png)

设置
---

1. [设置 Client ID & Secret](https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3266353)

2. [使用简悦官方的 Client ID & Secret](https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3266492)

3. [设置 Project ID](https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3266354)

4. [设置滴答清单或 Ticktick.com](https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3266356)

5. [设置标签](https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3266357)

用法
---

https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3277062

注意事项
---

1. 大部分插件都会在初次安装后自动弹出设置框，所有建议 **一个插件一个插件安装**。

2. 未避免配置文件互相覆盖，安装插件时仅打开选项页与随进入阅读模式的页面 e.g. https://sspai.com/post/69972

3. **仅安装此插件时需要关闭自动同步**，通过此插件安全其它插件时不需要关闭同步助手。

4. [授权错误 403](https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3289459)

更新日志
---

### 1.0.0

1. 可使用简悦官方提供的 **Client ID 与 Secret** 也可用户自行设置。

2. 可选择 **滴答清单** 或 **Ticktick.com**

3. 导入正文到滴答清单

4. 添加标签，并可使用稍后读定制的标签

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by Kenshin** [comment]
# 设置 Client ID & Secret

- [TickTick Developer](https://developer.ticktick.com/manage)

- [滴答清单](https://developer.dida365.com/manage/)

打开上述链接的一个，点击 **+ App Name**（下图均以 Ticktick.com 为例）

![image](https://user-images.githubusercontent.com/81074/181677990-7f3c0390-617b-4860-a2cd-716e875820dc.png)

输入名字（其它可忽略）后，点击 **Add**

![image](https://user-images.githubusercontent.com/81074/181678345-4019709c-cf0b-497b-ac04-2523c3cfb85c.png)

会看到下图所示的内容，点击 **Editor**

![image](https://user-images.githubusercontent.com/81074/181678459-1d5f48f0-df2c-472c-9387-2203cfeec5e6.png)

按下图所示，填入 `https://simpread.pro/auth_mini.html` 确认无误后，点击 **Save**

![image](https://user-images.githubusercontent.com/81074/181678762-aea40bb8-e128-49e7-a6c1-e69d0121376f.png)

至此，完成 Client ID & Secret 的生成，将上面的值复制到下图所示的位置

![image](https://user-images.githubusercontent.com/81074/181679016-571f9bd0-24da-40a3-9d15-a5c215f8e54b.png)

下图在阅读模式 → 右下角 → 触发器 → 导入到滴答清单选项

## 视频演示

https://user-images.githubusercontent.com/81074/181682209-586f6f58-3eab-4b64-aced-1b47ade72a1c.mp4

**Reply by Kenshin** [reply]
# 使用官方授权信息

如果不愿意自己动手生成，也可以使用简悦提供的值，按下图所示点击然后重新打开选项卡即可。

![image](https://user-images.githubusercontent.com/81074/181679753-12b18206-60ee-4efc-a419-fc4df671b128.png)

## 注意

此方案仅支持 Ticktick 账户，如果是滴答清单用户的话，还是需要自行设置 [Client ID & Secret](https://github.com/Kenshin/simpread/discussions/4281#discussioncomment-3266353)。

**Comment by Kenshin** [comment]
# Project ID

此值就是你在 Ticktick.com 上的清单 ID，如下图，点击 Reading List 得到的 URL 中端内容即是 Project ID

![image](https://user-images.githubusercontent.com/81074/181680296-6149ebb4-ba79-447e-b0a2-9545835ecdc2.png)

将此值复制到下图所示的位置，即可。

![image](https://user-images.githubusercontent.com/81074/181680490-cf357e74-62df-44cd-8590-231e216c6587.png)

**Comment by Kenshin** [comment]
# Region

导入到滴答清单支持 **滴答清单** 或 **Ticktick.com** 方案，在下图所示的位置切换。

![image](https://user-images.githubusercontent.com/81074/181680802-35b07013-3fd7-4f8f-817b-05e50f2c19ff.png)

# 注意

切换后对应的 Client ID & Secret 也需要对应，也就是说当选择了滴答清单，就应该在在 https://developer.dida365.com/manage/ 申请。

**Comment by Kenshin** [comment]
# 标签

可自行设置标签，同时也可以使用稍后读的标签

![image](https://user-images.githubusercontent.com/81074/181676504-4ab99a84-6724-43fc-9250-9f4a12a7d77a.png)

自行设置的标题，每行一个，导入到滴答清单后的效果

![image](https://user-images.githubusercontent.com/81074/181677806-86e7ba46-9035-4f1e-8a4d-b4483801b4cd.png)

**Comment by Kenshin** [comment]
# 基础用法

使用下图方式或快捷键 <kbd>t c</kbd> 导入

![image](https://user-images.githubusercontent.com/81074/181866508-791253bd-e7c2-480b-a00b-3cf5fab3d060.png)

# 高级用法

因为支持 **快捷键**，所以能跟 [自动化辅助增强](https://github.com/Kenshin/simpread/discussions/3596) 插件一起使用。

## 加入稍后读后自动导入到滴答清单

在自动化辅助插件中加入 <kdb>t c</kbd> 即可。

![image](https://user-images.githubusercontent.com/81074/181684274-1525a701-ba94-49b3-9b29-43954f73d1c5.png)

视频效果

https://user-images.githubusercontent.com/81074/181684143-7347a397-e406-4ba7-9746-9d8287c65292.mp4

**Comment by Joel2561** [comment]
# 授权错误 403

## 现象

![photo_2022-07-31_11-38-35](https://user-images.githubusercontent.com/77143461/182013707-8a329b3a-3e93-40cd-b893-20a2170453cd.jpg)

## 原因

1. 服务器不稳定、访问过于频繁、ip被ban、DNS解析问题等；
2. **身份验证错误**。

## 解决办法

1. 稍后再试、重建DNS缓存等；
2. 注意自己使用的账户是**滴答清单的账户**（国内）还是**ticktick的账户**（国外），在简悦插件中选择好**对应的账户**（如图）。
![image_2022-07-31_13-49-50](https://user-images.githubusercontent.com/77143461/182013747-8cb5552e-87ab-4291-ab97-53fadcbfb27c.png)

**Comment by AirMax121** [comment]
是的 标题会自动读取网页标题  我这边网络不好一直没显示出来  刚关了vpn正常了  评论都删了没想到老大这么快看到 哈哈

**Reply by Kenshin** [reply]
哈，小事情~ 😂

**Comment by iblogc** [comment]
讨论内容转移到 https://github.com/Kenshin/simpread/issues/4601
